game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -70,393 +64,182 @@ public ConcurrentBattleCalculator(final String threadNamePrefix) {
-      // see how long 1 copy takes (some games can get REALLY big)
-      final long startTime = System.currentTimeMillis();
-      final long startMemory =
-          Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
-      final GameData newData;
-      try {
-        // make first copy, then release lock on it so game can continue (ie: we don't want to lock
-        // on it while we copy
-        // it 16 times, when once is enough) don't let the data change while we make the first copy
-        data.acquireWriteLock();
-        newData = GameDataUtils.cloneGameDataWithoutHistory(data, false);
-      } finally {
-        data.releaseWriteLock();
-      }
-      currentThreads = getThreadsToUse((System.currentTimeMillis() - startTime), startMemory);
-      try {
-        // make sure all workers are using the same data
-        newData.acquireReadLock();
-        int i = 0;
-        // we are already in 1 executor thread, so we have MAX_THREADS-1 threads left to use
-        if (currentThreads <= 2 || MAX_THREADS <= 2) {
-          // if 2 or fewer threads, do not multi-thread the copying (we have already copied it once
-          // above, so at most
-          // only 1 more copy to make)
-          while (cancelCurrentOperation.get() >= 0 && i < currentThreads) {
-            // the last one will use our already copied data from above, without copying it again
-            workers.add(new BattleCalculator(newData, (currentThreads == ++i)));
-          }
-        } else { // multi-thread our copying, cus why the heck not (it increases the speed of
-          // copying by about double)
-          final CountDownLatch workerLatch = new CountDownLatch(currentThreads - 1);
-          while (i < (currentThreads - 1)) {
-            ++i;
-            executor.execute(
-                () -> {
-                  if (cancelCurrentOperation.get() >= 0) {
-                    workers.add(new BattleCalculator(newData, false));
-                  }
-                  workerLatch.countDown();
-                });
-          }
-          // the last one will use our already copied data from above, without copying it again
-          workers.add(new BattleCalculator(newData, true));
-          Interruptibles.await(workerLatch);
-        }
-      } finally {
-        newData.releaseReadLock();
-      }
-    }
-    if (cancelCurrentOperation.get() < 0 || data == null) {
-      // we could have cancelled while setting data, so clear the workers again if so
-      workers.clear();
-      isDataSet = false;
-    } else {
-      // should make sure that all workers have their game data set before we can call calculate and
-      // other things
-      isDataSet = true;
-      dataLoadedAction.run();
-    }
-    // allow setting new data to take place if it is waiting on us
-    latchWorkerThreadsCreation.countDown();
-    // allow calcing and other stuff to go ahead
-    latchSetData.countDown();
+    return MAX_THREADS;
```

Comment:
side-note, it seems questionable to return MAX_THREADs and to also use that same constant in this class. It looks like this 'get' is only used for display purposes. When multi-threading of the battle calc was brand new, made more sense. Now I'm more thinking it's an implementation detail that the calc is multi-threaded and we don't really need to tell the user that (it's not important to them, only that the calc completes quickly and accurately).  CC: @ron-murhammer 

In that case, if we remove that, we could remove this interface method and implementation.

Reply:
Good Point. It sounds like something #6329 should include, so I'd rather make a change there and merge this PR as-is.

This count is currently used for the loading screen where it says "Calculating Odds (X Threads)" or something similar. I think displaying the thread count is a good explanation/excuse for the high CPU usage the "BattleCalculator" uses. (My Laptop has "only" 4 cores, so it's definitely noticeable that everything gets slower once the simulation runs).
That's why I wonder: Do you think it's a reasonable solution to create a class that provides the amount of available threads and make the UI "assume" that the actual implementation is using the same amount of cores?

Reply:
In some sense we're already relying on java to actually use 4 different threads for our battlecalc (there's no guarantee the threads will all be run side by side at the same time)

Reply:
Personally, the "X threads" is interesting, but IMO more for devs. A good percentage of users will not know what a thread even is.

Reply:
> Good Point. It sounds like something #6329 should include, so I'd rather make a change there and merge this PR as-is.

Indeed, to be explicit, any comment I prefix with "side-note" is really just a general comment; My general intent is the item is something that should *not* be addressed in this PR, but is worth a mention maybe and a follow-up.